### PR TITLE
new topic about troubleshooting

### PR DESCRIPTION
**Pull request by FredHan001**
According to [this github issue](https://github.com/dotnet/AspNetCore.Docs/issues/19570), I'd like to submit this troubleshooting guidance for common Signalr WebSocket handshake issues.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=20056) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=20056) before we can accept your contribution.<br/><br/>:x: FredHan001 [sign now](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=20056)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/AspNetCore.Docs?pullRequest=20056) it.</sub>

File: aspnetcore/signalr/javascript-client.md [hunk 1]
```diff
@@ -262,6 +262,41 @@ The following code demonstrates a typical manual reconnection approach:
 
 A real-world implementation would use an exponential back-off or retry a specified number of times before giving up.
 
+## Troubleshoot the error that occurs during WebSocket handshake
+
+A *"Error during WevSocket handshake"* exception that occurs when we try to establish connection to ASP.NET Core SignalR hub can be diagnosed using the advice in this topic.
```
Comment (bradygaster) [inline aspnetcore/signalr/javascript-client.md:267]: WebSocket - typo

**Review by bradygaster — APPROVED** [review]

**Comment by BrennanConroy** [comment]
I think we need a new page: "Troubleshooting connection issues" and contain more general information about possible errors.
These errors aren't neccesarily specific to the javascript client.

File: aspnetcore/signalr/javascript-client.md [hunk 2]
```diff
@@ -262,6 +262,46 @@ The following code demonstrates a typical manual reconnection approach:
...
+```log
+WebSocket connection to 'ws://xxx/HubName' failed: Error during WebSocket handshake: Unexpected response code: 307
+```
+
+Frequently this occurs when the SignalR hub server:
+
+* Listens to and responds over both HTTP and HTTPS.
+* Is configured to enforce HTTPS by calling `UseHttpsRedirection` in `Startup`, or enforces HTTPS via URL rewrite rule.
+
+This error can be caused by specifying the HTTP URL on client side using `.withUrl("http://xxx/HubName")`. The fix for this case is modifying the code to use HTTPS URL or not enforcing HTTPS.
```
Comment (Rick-Anderson) [inline aspnetcore/signalr/javascript-client.md:292]: @blowdart  I'm uncomfortable telling customers to not enforce HTTPS. Can I remove ~or not enforcing HTTPS~ ?

**Comment by Rick-Anderson** [comment]
> I think we need a new page: "Troubleshooting connection issues" and contain more general information about possible errors.
> These errors aren't necessarily specific to the javascript client.

@BrennanConroy  should I open a new issue for that?  Can you review this PR or do you want to hold up until we add the new page

File: aspnetcore/signalr/javascript-client.md [hunk 3]
```diff
@@ -262,6 +262,46 @@ The following code demonstrates a typical manual reconnection approach:
 
 A real-world implementation would use an exponential back-off or retry a specified number of times before giving up.
 
+## Troubleshoot WebSocket handshake errors
+
+This section provides help on the *"Error during WevSocket handshake"* exception that occurs when trying to establish connection to ASP.NET Core SignalR hub.
```
Comment (bradygaster) [inline aspnetcore/signalr/javascript-client.md:267]: "WebSocket"

File: aspnetcore/signalr/javascript-client.md (hunk 2 above)
Comment (bradygaster) [reply aspnetcore/signalr/javascript-client.md:292]: cool with that

**Review by bradygaster — APPROVED** [review]
per my and barry's comments

**Comment by Rick-Anderson** [comment]
@FredHan001 thanks for the valuable PR. Are you a MS employee?

**Comment by BrennanConroy** [comment]
> @BrennanConroy should I open a new issue for that? Can you review this PR or do you want to hold up until we add the new page

Eh, have feedback but we can do it when the new page is added.

File: aspnetcore/signalr/javascript-client.md (hunk 2 above)
Comment (blowdart) [reply aspnetcore/signalr/javascript-client.md]: Yea, that's naughty, cull it :)

**Pull request merged**